backend-tests/tests/test_rbac.py

```diff
@@ -0,0 +1,257 @@
+
+                device.group = group
+                grouped_devices[group].append(device)
+                tenant.devices.append(device)
+
+        return grouped_devices
+
+    def add_user_to_role(self, mgo_client, user, tenant_id, roles):
+        """
+        Ensures that the role exists and adds the user to it.
+        NOTE: if creating a role to limit deployable groups, make sure to add
+              another role that permits usage of the API endpoint for creating
+              deployments.
+        :param mgo_client: (pymongo.MongoClient)
+        :param user:       the user to which the roles should be added
+                           (common.User)
+        :param tenant_id:  the tenant for which the user belongs (str)
+        :param roles:      the (list) of roles to constrain the user
+                           (UserRole or list{UserRole})
+        :return: None
+        """
+        useradm_MGMT = ApiClient(useradm.URL_MGMT)
+        if getattr(user, "token", None) is None:
+            rsp = useradm_MGMT.call(
+                "POST", useradm.URL_LOGIN, auth=(user.name, user.pwd)
+            )
+            assert rsp.status_code == 200, rsp.text
+            user.token = rsp.text
+
+        if not isinstance(roles, list):
+            roles = [roles]
+        for role in roles:
+            try:
+                coll_roles = getattr(mgo_client, "useradm-%s" % tenant_id).roles
+                coll_roles.insert_one(role.dict)
+            except pymongo.errors.DuplicateKeyError:
+                # Role already exists
+                pass
+
+        rsp = useradm_MGMT.with_auth(user.token).call(
+            "PUT",
+            useradm.URL_USERS_ID.format(id=user.id),
+            body={"roles": [role._id for role in roles]},
+        )
+        assert rsp.status_code == 204, rsp.text
+
+    @pytest.mark.parametrize(
+        "test_case",
+        [
+            {
+                "name": "Test RBAC deploy to device group",
+                "user": {"name": "test1@username.org", "pwd": "password"},
+                "permissions": [
+                    UserPermission("CREATE_DEPLOYMENT", "DEVICE_GROUP", "test")
+                ],
+                "device_groups": {"test": 5, "production": 30, "staging": 20},
+                "deploy_groups": ["test"],
+                "status_code": 201,
+            },
+            {
+                "name": "Test RBAC deployment forbidden",
+                "user": {"name": "test2@username.org", "pwd": "password"},
+                "permissions": [
+                    UserPermission("CREATE_DEPLOYMENT", "DEVICE_GROUP", "test")
+                ],
+                "device_groups": {"test": 5, "production": 25},
+                "deploy_groups": ["production"],
+                "status_code": 405,
+            },
+            # { TODO: This test-case should pass, but deployments only
+            #         expect device IDs to come from the same group
```

Comment:
yes, thats true, until we get MEN-3411 in. and when deployments restrictions are on with RBAC there is no other way: but to restrict the deployment constructor to only one group.

Reply:
I'll just leave this test case commented out until this is implemented.